### PR TITLE
homme SYCL changes

### PR DESCRIPTION
**Pull request by oksanaguba**
Changes needed for spot/aurora: sycl kokkos spaces, temporary TP changes (TP init will be sorted out later), switch booleans to ints in F<-->xx interfaces.

see #6569 for the original branch/history.

[bfb]

### CODE REVIEW

File: components/homme/src/theta-l_kokkos/cxx/HyperviscosityFunctorImpl.cpp [hunk 1]
```diff
@@ -118,9 +118,9 @@ void HyperviscosityFunctorImpl::init_params(const SimulationParams& params)
   m_eos.init(params.theta_hydrostatic_mode,m_hvcoord);
 
 #ifdef HOMMEXX_BFB_TESTING
-  m_process_nh_vars = true;
+  m_process_nh_vars = 1;
 #else
-  m_process_nh_vars = !params.theta_hydrostatic_mode;
+  m_process_nh_vars = static_cast<int> (not params.theta_hydrostatic_mode);
```
Comment (oksanaguba) [inline components/homme/src/theta-l_kokkos/cxx/HyperviscosityFunctorImpl.cpp:123]: Luca already made this change, so i won't revert to bool.

**Comment by ambrad** [comment]
As I understand it, the int/bool code is just to work around a compiler bug affecting the F90-C++ interface; essentially, the compiler's treatment of c_bool is broken, or something like that. In particular, using bool in a kernel is _not_ an issue; we do that in several spots. Oksana, is all of that correct?

If so, I can add a commit to this branch next week that isolates the workaround to just the F90-C++ interface code. In particular, I'd like to remove the two theta_hydrostatic_mode variables from control_mod.F90. Is that OK?

**Comment by oksanaguba** [comment]
> As I understand it, the int/bool code is just to work around a compiler bug affecting the F90-C++ interface; essentially, the compiler's treatment of c_bool is broken, or something like that. In particular, using bool in a kernel is _not_ an issue; we do that in several spots. Oksana, is all of that correct?

Yes, correct about int-bool issue.

> If so, I can add a commit to this branch next week that isolates the workaround to just the F90-C++ interface code. In particular, I'd like to remove the two theta_hydrostatic_mode variables from control_mod.F90. Is that OK?

yes, thanks!

**Comment by rljacob** [comment]
@bartgol thank for reducing the number of commits but you could do this one more time using --author on the commit command to give author credit to Oksana?

**Comment by bartgol** [comment]
Of course. I'll do it as soon as I get back to my computer

**Comment by bartgol** [comment]
@rljacob @oksanaguba I edited the commits, to fix the author.

@ambrad FYI, in case you pulled and was planning to do the mods you mentioned...

**Review by bartgol — APPROVED** [review]

**Comment by ambrad** [comment]
I think I'll have a commit to push to this branch by tonight. I've made changes and am running homme_integration on Chrysalis.

**Comment by ambrad** [comment]
Done. I also fixed some moisture-related code in preqx so that homme_integration passes on Chrysalis.

**Review by ambrad — APPROVED** [review]

**Review by bartgol — APPROVED** [review]

**Review by bartgol — APPROVED** [review]

**Review by ambrad — APPROVED** [review]

**Comment by oksanaguba** [comment]
homme suite pass on chrysalis

**Comment by oksanaguba** [comment]
in next

**Comment by oksanaguba** [comment]
hommexx ers_d test failed on cdash with an mpi error, but i ran it by hand with no issues.

**Pull request merged**